wire-ios/Wire-iOS/Sources/Helpers/SemanticColors.swift

```diff
@@ -269,39 +269,16 @@ enum SemanticColors {
 }
 
 extension UIColor {
-    convenience init(light: ColorAsset, dark: ColorAsset) {
+
+    convenience init(light: ColorResource, dark: ColorResource) {
         self.init { traits in
-            return traits.userInterfaceStyle == .dark ? dark.color : light.color
+            .init(resource: traits.userInterfaceStyle == .dark ? dark : light)
         }
     }
-}
-
-extension UIColor {
-    var swiftUIColor: Color {
-        return Color(uiColor: self)
-    }
 
-    convenience init(for accentColor: WireAccentColor) {
-        switch accentColor {
-        case .blue:
-            self.init(light: Asset.Colors.blue500Light, dark: Asset.Colors.blue500Dark)
-        case .green:
-            self.init(light: Asset.Colors.green500Light, dark: Asset.Colors.green500Dark)
-        case .yellow: // Deprecated
-            self.init(red: 0.996, green: 0.749, blue: 0.007, alpha: 1)
-        case .red:
-            self.init(light: Asset.Colors.red500Light, dark: Asset.Colors.red500Dark)
-        case .amber:
-            self.init(light: Asset.Colors.amber500Light, dark: Asset.Colors.amber500Dark)
-        case .turquoise:
-            self.init(light: Asset.Colors.turquoise500Light, dark: Asset.Colors.turquoise500Dark)
-        case .purple:
-            self.init(light: Asset.Colors.purple500Light, dark: Asset.Colors.purple500Dark)
+    convenience init(light: ColorAsset, dark: ColorAsset) {
+        self.init { traits in
+            traits.userInterfaceStyle == .dark ? dark.color : light.color
```

Comment:
One next step could be to use Xcode asset catalog directly, because it supports dark mode out of the box.

Reply:
I'm not sure the mapping is 1:1 for color uses. 
Text for example could be black in light mode, but grey in dark mode. This might not apply to view backgrounds.